prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionBehavior.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    public class SelectionBehavior : DependencyObject
+    {
+
+
+
+        public bool IsSelected
+        {
+            get { return (bool)GetValue(IsSelectedProperty); }
+            set { 
+                SetValue(IsSelectedProperty, value);
+                IsSelectedChanged(value);
+            }
+        }
+
+        // Using a DependencyProperty as the backing store for IsSelected.  This enables animation, styling, binding, etc...
+        public static readonly DependencyProperty IsSelectedProperty =
+            DependencyProperty.Register("IsSelected", typeof(bool), typeof(SelectionBehavior), new PropertyMetadata(false));
+
+
+
+        private readonly Control element;
+
+        public SelectionBehavior(Control element)
+        {
+            if(element == null)
+            {
+                return;
+            }
+            this.element = element;
+            VisualStateManager.GoToState(element, "Deselected", true);
+        }
+        private void IsSelectedChanged(bool value)
+        {
+            if(value)
+            {
+                VisualStateManager.GoToState(element, "Selected", true);
```

Comment:
The issue I foresee here with this approach is that we cannot use a single element in the visual tree across these behaviors. For example, if I had one border that i want to change color in pointer over and re-use the same thing to highlight selection, It is likely to cause problems depending on the order in which we attach the behaviors. The advantage of doing that is reducing the number of elements in the tree.

Reply:
Yes, I've updated both behaviors to just expose properties that we can use to update the states (using Triggers) or what ever suites the users best.

Reply:
Neat. I like the idea of removing the goto state in the behavior. 👍